hawc/apps/common/lookups.py

```diff
@@ -60,9 +60,24 @@ def get_underscore_field_val(self, obj: Any, underscore_path: str) -> Any:
         Returns:
             Any: the desired attribute of the object or child object.
         """
+
+        attr_whitelist = [
+            "endpoint",
+            "animal_group",
+            "experiment",
+            "name",
+            "created",
+            "last_updated",
+            "data_type",
+            "response_units",
+            "observation_time",
+            "system",
+        ]
         obj_ = obj
         try:
             for attr in underscore_path.split("__"):
+                if attr not in attr_whitelist:
```

Comment:
RelatedLookup is used in lots of places, but we only allow a user provided query string in `EndpointByAssessmentLookup`. 

It might be best to create a new subclass of `RelatedLookup` called `UserProvidedRelatedLookup` that has this additional check?

Also, I'd probably pull this check out of the try/except and put it up above.  Something like:

```python
        attributes = underscore_path.split("__")
        if not all(attr in self.attr_whitelist for attr in attributes):
            raise AttributeError("Access to {underscore_path} is not allowed.")

        try:
            for attr in attributes:
                obj_ = getattr(obj_, attr)
        except AttributeError:
            raise AttributeError(f"Element {underscore_path} not found in {obj}")

```

Reply:
Thinking about this more, might be best to add a `get_search_path` method that receives the user request, and does our whitelist validation check upstream without messing with the query

Reply:
I'm confused. How does adding `get_search_path` make sure it's validated before `get_query`? would that method be called inside `get_query`?